packages/attribute-slicer-powerbi/src/AttributeSlicerVisual.ts

```diff
@@ -305,18 +305,24 @@ export default class AttributeSlicer extends VisualBase {
                     delete this.loadDeferred;
                 }
 
-                const columnName = ldget(dv, "categorical.categories[0].source.queryName");
+                const columnNames: Array<String> = [];
+                _.forOwn(ldget(dv, "categorical.categories"), (value, key: any) => {
+                    columnNames.push(value.source.queryName);
+                });
 
                 // Only clear selection IF
                 // We've already loaded a dataset, and the user has changed the dataset to something else
-                if (this.currentCategory && (!columnName || this.currentCategory !== columnName)) {
+                if (this.currentCategory && (!columnNames || !_.isEqual(this.currentCategory, columnNames)))  {
```

Comment:
The `!columnNames` will always be false (since it is an array), so you can probably remove it.